enhancements/machine-api/out-of-tree-provider-support.md

```diff
@@ -0,0 +1,182 @@
+---
+title: neat-enhancement-idea
+authors:
+  - "@Danil-Grigorev"
+reviewers:
+  - "@enxebre"
+  - "@elmiko"
+  - "@mfedosin"
+approvers:
+  - "@enxebre"
+  - "@elmiko"
+  - TBD
+creation-date: 2020-08-31
+last-updated: 2020-09-01
+status: implementable
+see-also:
+  - "/enhancements/cloud-controller-manager/openstack-cloud-controller-manager.md"  
+replaces:
+superseded-by:
+---
+
+# Add support for out-of-tree cloud provider integration
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+1) Should we running `cloud-controller-manager` as:
+- static `Pod` managed from the `Node` filesystem.
+- `Deployment` backing up the `Pod`.
+- `ReplicaSet` to run a `Pod` on each master node.
+
+## Summary
+
+This enhancement proposal describes steps needed to add support for out-of-tree providers in cloud providers currently supported by Openshift. This includes: AWS, Azure, GCP, vSphere, IBM. Openstack provider proposal is located at it's dedicated document.
```

Comment:
Indeed, because doesn't the OpenStack document propose that they will implement this as part of supporting the OpenStack provider?  Is this enhancement meant to provide the details needed?

Right now it looks like the two enhancements should be merged into one.  #423

Reply:
We merged these two together here :+1: